packages/server/message-store-api/service/MessageStoreService.go

```diff
@@ -226,26 +285,20 @@ func (s *MessageService) getMessagePreview(input *msProto.InputMessage) string {
 	return previewContent
 }
 
-func (s *MessageService) getInitiatorId(st msProto.SenderType, tenant string, email string) (*string, error) {
-	var initiatorId *string
-	if st == msProto.SenderType_CONTACT {
-		if contact, err := s.customerOSService.GetContactWithEmailOrCreate(tenant, email); err != nil {
-			return nil, err
-		} else {
-			initiatorId = &contact.Id
-		}
-	} else if st == msProto.SenderType_USER {
-		if user, err := s.customerOSService.GetUserByEmail(email); err != nil {
+func (s *MessageService) getParticipant(tenant string, initiatorIdentifier string) (*Participant, error) {
+	user, err := s.customerOSService.GetUserByEmail(initiatorIdentifier)
+	if err != nil {
+		contact, err := s.customerOSService.GetContactWithEmailOrCreate(tenant, initiatorIdentifier)
+		if err != nil {
 			return nil, err
-		} else {
-			initiatorId = &user.Id
 		}
+		return &Participant{Id: contact.Id, Type: entity.CONTACT}, nil
+	} else {
+		return &Participant{Id: user.Id, Type: entity.CONTACT}, nil
 	}
-
-	return initiatorId, nil
 }
 
-func (s *MessageService) saveConversationEvent(tenant string, conversation *Conversation, input *msProto.InputMessage, initiatorId string, senderType entity.SenderType) entity.ConversationEvent {
+func (s *MessageService) saveConversationEvent(tenant string, conversation *Conversation, input *msProto.InputMessage, participant *Participant) entity.ConversationEvent {
```

Comment:
variable name is confusing, it should remain called sender despite being of type Participant, as it is being used to indicate which participant sent the message

Reply:
Fixed